plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/DoubleDecoder.java

```diff
@@ -29,6 +29,10 @@ public void decode(Supplier<Object> getter, BlockBuilder output)
         if (value == null) {
             output.appendNull();
         }
+        else if (value instanceof String) {
+            // Pinot returns NEGATIVE_INFINITY, POSITIVE_INFINITY as a String
+            DOUBLE.writeDouble(output, Double.valueOf(value.toString()));
+        }
         else {
             DOUBLE.writeDouble(output, ((Number) value).doubleValue());
```

Comment:
Seeing a cast to `Number` makes me sad - can we make this stricter?

Reply:
This code was unchanged from before, the reason: pinot transform functions declare that they return double (for example) but the actual value can be a double, long, int, or float. Let me know if you have any suggestions.

Reply:
The problem with Number is that it applies to things like BigDecimal too.
We can address this separately as part of larger effort to remove Number casts from the code where possible.

Let's keep as is for now.